Content.Server/Explosion/EntitySystems/ExplosionSystem.cs

```diff
@@ -255,6 +260,40 @@ public float IntensityToRadius(float totalIntensity, float slope, float maxInten
 
         QueueExplosion(mapPos, typeId, totalIntensity, slope, maxTileIntensity, tileBreakScale, maxTileBreak, canCreateVacuum, addLog: false);
 
+        if (user == null)
+        {
+            _gameTicker.RecordReplayEvent(new ReplayExplosionEvent()
+            {
+                Severity = ReplayEventSeverity.High,
+                Time = _gameTiming.CurTick.Value,
+                EventType = ReplayEventType.Explosion,
+                Intensity = totalIntensity,
+                Slope = slope,
+                MaxTileIntensity = maxTileIntensity,
+                TileBreakScale = tileBreakScale,
+                MaxTileBreak = maxTileBreak,
+                CanCreateVacuum = canCreateVacuum,
+                Type = typeId
+            });
+        }
+        else
+        {
+            _gameTicker.RecordReplayEvent(new ReplayExplosionEvent()
+            {
+                Severity = ReplayEventSeverity.High,
+                Time = _gameTiming.CurTick.Value,
+                EventType = ReplayEventType.Explosion,
+                Source = _gameTicker.GetPlayerInfo(user.Value),
+                Intensity = totalIntensity,
+                Slope = slope,
+                MaxTileIntensity = maxTileIntensity,
+                TileBreakScale = tileBreakScale,
+                MaxTileBreak = maxTileBreak,
+                CanCreateVacuum = canCreateVacuum,
+                Type = typeId
+            });
+        }
+
```

Comment:
This is awful

Reply:
I agree, but it's a draft. I am still working on it.